### DISCUSSION TITLE
Search Confusion

### DISCUSSION THREAD

**Discussion by wind57** [provectus/kafka-ui#2579] [Q&A] [answered]
If I want to search for a certain message and go to the search box and input either:

`value.id==507566268` or `id==507566268`, I get zero results.

On the other hand, if I create a _new_ filter and input the same code under "filter code": `value.id==507566268`, I do get the result. :| 

How come?

One more question - what is the proper syntax for an array search? Cause nothing seems to work for me. I hoped:

`users.foods[].id=123` would work.

**Comment by Haarolean** [comment, marked as answer]
Hi, please use discussions for questions instead of issues.

Search box is a simple search which works checks if either the key or the message contains the string from the input.

Filters are smart on the other hand, that's why they're called smart filters.

Regarding the array search, I don't think you can do something exact like this with groovy. You'll have to write a for loop instead.
But it's a good idea to support things like this via json paths, I've created an issue -- #2580

**Comment by wind57** [comment]
> Hi, please use discussions for questions instead of issues

my bad. will do next time.

> You'll have to write a for loop instead

can you point me to a minor sample please?

**Reply by Haarolean** [reply]
![image](https://user-images.githubusercontent.com/1494347/190285652-243ca66f-b90b-46ca-8f5e-e108d5882205.png)

**Comment by wind57** [comment]
> Search box is a simple search which works checks if either the key or the message contains the string from the input

that seems to work rather weird too :/

If I search for `"name"` -> it works. 
If I search for `"wind57"` -> it works.

If I search for `"name": "wind57"` -> it does not. And of course such an entry exists

**Comment by wind57** [comment]
ah yes, you need to search _without_ a space, pretty "obvious" :)

**Reply by iliax** [reply]
@wind57 I am not sure you using suggested code snipped properly.
filter is **any** valid groovy expression with `value, key, valueAsText, keyAsText, timestampMs` variables set.
here is sample (NOTE: this is current **master** image):

Messages in topic: 
<img width="1408" alt="Screenshot 2022-09-15 at 16 01 54" src="https://user-images.githubusercontent.com/702205/190399515-9a4508f2-37db-4fde-a0fc-9972f851955a.png">

------------

sample filters (both work as expected):


<img width="562" alt="Screenshot 2022-09-15 at 16 04 14" src="https://user-images.githubusercontent.com/702205/190399582-f0ef1dfe-cab1-4812-8d94-25fd0b5dbf12.png">

-------------

<img width="557" alt="Screenshot 2022-09-15 at 16 05 00" src="https://user-images.githubusercontent.com/702205/190399599-7bbdee49-91bf-44e4-95ab-829e16f68080.png">

**Reply by wind57** [reply]
@iliax I was referring in my last comment:

> ah yes, you need to search without a space, pretty "obvious" :)

about the _non_ smart filter part. This issue was resolved for me. 

---

The _other_ issue I had was with groovy syntax, but it seems to again work for me now against a different payload. So overall, I think that everything is just fine. I'll ask again if I find something that does not work. thank you for your time.
